src/Data/Map/Internal.purs

```diff
@@ -612,6 +613,11 @@ union = unionWith const
 unions :: forall k v f. Ord k => Foldable f => f (Map k v) -> Map k v
 unions = foldl union empty
 
+-- | Difference of two maps. Return elements of the first map where
+-- | the keys do not exist in the second map.
+difference :: forall k v. Ord k => Map k v -> Map k v -> Map k v
```

Comment:
If the values of the second map aren't used then this could be typed as:
`forall k v w . Ord k => Map k v -> Map k w -> Map k v`

Reply:
But should it be? I think the type as it is makes sense for a binary minus operator.
Then you can't by accident subtract a `Map String Int` from a `Map String String`, holes can help you find a matching pair, etc.

Someone previously made a similar choice for `unionWith :: forall k v. Ord k => (v -> v -> v) -> Map k v -> Map k v -> Map k v` which could well be `unionWith :: forall k a b c. Ord k => (a -> b -> c) -> Map k a -> Map k b -> Map k c`.

(And I think the case for generalizing `unionWith` is much stronger.)